packages/frontend/src/hooks/useAddTokenToMetamask.ts

```diff
@@ -17,9 +17,14 @@ export function useAddTokenToMetamask(
   const [success, setSuccess] = useState<boolean>(false)
 
   const addToken = useCallback(
-    (networkId?: number) => {
+    (networkId: number) => {
       if (provider && token) {
-        const { symbol, image, decimals } = token
+        let { symbol, image, decimals } = token
+
+        if (symbol === 'XDAI' && networkId !== 100) {
```

Comment:
maybe it'd be better to compare against the slug here instead of hardcoded network number

```ts
Chain.xDai !== networkIdToSlug(networkId)
```